guides/runtime-parser.md

```diff
@@ -0,0 +1,319 @@
+const APP_SHELL_URL = '/home';
+const INLINE_IMAGES: string[] = ['png', 'svg', 'jpg'];
+const ROUTE_DEFINITIONS = [
+      '/home',
+      '/about/:id'
+    ];
+
+self.addEventListener('install', function (event: any) {
+  const parser = System.import('@angular/app-shell')
+    .then((m: any) => {
+      context.ngShellParser = m.shellParserFactory({
+        APP_SHELL_URL,
+        ROUTE_DEFINITIONS,
+        SHELL_PARSER_CACHE_NAME,
+        INLINE_IMAGES
+      });
+      return context.ngShellParser;
+    })
+    .then(() => context.ngShellParser.fetchDoc())
+    .then((res: any) => context.ngShellParser.parseDoc(res))
+    .then((strippedResponse: any) => {
+      return (<any>self).caches.open(SHELL_PARSER_CACHE_NAME)
+        .then((cache: any) => {
+          strippedResponse
+            .clone()
+            .text()
+            .then((txt: string) => console.log(txt));
+          return cache.put(APP_SHELL_URL, strippedResponse);
+        });
+    })
+    .catch((e: any) => console.error(e));
+  event.waitUntil(parser);
+});
+
+self.addEventListener('fetch', function (event: any) {
+  event.respondWith(
+    context.ngShellParser.match(event.request)
+      .then((response: any) => {
+        if (response) return response;
+        return (<any>self).caches.match(event.request)
+          .then((response: any) => {
+            if (response) return response;
+            return (<any>self).fetch(event.request);
+          })
+      })
+  );
+});
+```
+
+Since the logic in the snippet above is not trivial, lets explore it step-by-step:
+
+## Step 1 - Configuration
+
+```typescript
+const SHELL_PARSER_CACHE_NAME = 'app-shell:cache';
+const APP_SHELL_URL = '/home';
+const INLINE_IMAGES: string[] = ['png', 'svg', 'jpg'];
+const ROUTE_DEFINITIONS = [
+      '/home',
+      '/about/:id',
+    ];
+```
+
+First, we declare a constant called `SHELL_PARSER_CACHE_NAME`. This is the name of the cache where we are going to store the App Shell's template once it has been generated.
+
+The `APP_SHELL_URL` is the URL of the view which template we are going to use in order to generate the App Shell.
+
+Since our final goal is to render the entire App Shell as quickly as possible, we want to inline all the referenced within its elements resources. The next step of our declarations is the `INLINE_IMAGES` array. It provides a list of of image extensions that we want to be inlined as base64 strings.
+
+Finally, we declare a list of routes that we want to be handled by the parser. For instance, once the user visits the `/home` route the first thing that we want to do is to render the cached App Shell. Right after the root component of the route has been initialized and all the associated to it external resources are available, its content will be rendered on the place of the App Shell. As we can see from the route `/about/:id`, the `ROUTE_DEFINITIONS` supports wildcards, similar to the parameters in the Angular's router.
```

Comment:
Type: _in the Angular's router_  --> _in Angular's router_ (or _in the Angular router_)